metals/src/main/scala/scala/meta/internal/metals/ForwardingMetalsBuildClient.scala

```diff
@@ -174,4 +183,50 @@ final class ForwardingMetalsBuildClient(
       case _ =>
     }
   }
+
+  def ongoingCompilations: Array[TreeViewNode] = {
+    compilations.keysIterator.flatMap(ongoingCompileNode).toArray
+  }
+
+  def ongoingCompileNode(
```

Comment:
I'm not sure the presentation logic should be held together with the "business" logic.